casper/src/main/scala/coop/rchain/casper/Validate.scala

```diff
@@ -361,6 +363,15 @@ object Validate {
     }
   }
 
+  def deployCount[F[_]: Applicative: Log](b: BlockMessage): F[Either[InvalidBlock, ValidBlock]] =
+    if (b.header.get.deployCount == b.body.get.deploys.length) {
```

Comment:
We validate these fields are populated first before calling `.get` on them right? We really should encode whether those field checks happened or not into the type...

Reply:
Yes and I agree

Reply:
Yes, I remember that we had a discussion on #rholang about this and the proposed solution was to write a macro which would automatically generate safe protobuf classes (which do not wrap body and header into Option for example).

I am wondering whether using `required` for header and body were considered... There are a few drawbacks to this approach, but this might make sense.

Reply:
Oh, I didn't know it was removed. NVM then.